### ISSUE TITLE
[BUG] [scala-sttp] api.mustache template specifies the incorrect return type

### ISSUE THREAD

**Issue by reddicj** [OpenAPITools/openapi-generator#13077]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [x] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

The return type of each operation is incorrect. Currently the template defines it as this:

```
def {{operationId}}({{>methodParameters}}): Request[{{#separateErrorChannel}}Either[ResponseException[String, Exception], {{>operationReturnType}}]{{/separateErrorChannel}}{{^separateErrorChannel}}{{>operationReturnType}}{{/separateErrorChannel}}, Nothing] =
```

But it should be this:

```
def {{operationId}}({{>methodParameters}}): Request[{{#separateErrorChannel}}Either[ResponseException[String, Exception], {{>operationReturnType}}]{{/separateErrorChannel}}{{^separateErrorChannel}}{{>operationReturnType}}{{/separateErrorChannel}}, Any] =
```

i.e. `Nothing` should be `Any`

##### openapi-generator version

6.0.1

##### Generation Details

Using Coursier launch:
```
 cs launch org.openapitools:openapi-generator-cli:6.0.1 -- generate \
  -i api.yaml \
  -g scala-sttp \
  -o out
```

##### Steps to reproduce

See the sttp docs:
https://sttp.softwaremill.com/en/latest/requests/type.html

##### Suggest a fix

Suggested above.

**Comment by reddicj** [comment]
I see this issue is fixed in this PR #11949.

**Comment by fserra-mdsol** [comment]
This has not been fixed yet, as the PR with the fix has not been merged
